### PR TITLE
Mobile selection change

### PR DESCRIPTION
**Pull request by nightwing**
A small fix to allow moving cursor by swiping spacebar on android.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/keyboard/textinput.js [hunk 1]
```diff
@@ -270,6 +274,20 @@ var TextInput = function(parentNode, host) {
...
     };
 
+    function detectSelectionChange(e) {
+        if (!text || !text.parentElement)
+            document.removeEventListener("selectionchange", detectSelectionChange);
+
+        var startDiff = text.selectionStart - lastSelectionStart;
+        if (text.selectionEnd == text.selectionStart && startDiff && !inComposition) {
+            var repeat = Math.abs(startDiff);
+            var key = repeat == startDiff ? KEYS.right : KEYS.left;
```
Comment (InspiredGuy) [inline src/keyboard/textinput.js:284]: same here

**Review by InspiredGuy — COMMENTED** [review]
could you please add tests for new functionality?

**Comment by InspiredGuy** [comment]
is this fix related to this issue https://github.com/ajaxorg/ace/issues/4178? also, is there a reason to add this functionality to android-only, is behaviour on iOS significantly different?

**Comment by nightwing** [comment]
It is indeed related, i remembered that there was a report, but thought it got closed by the autoclose bot. 
The issue with ios is that there is another [code](https://github.com/ajaxorg/ace/blob/a5cc588e8ee90d036ca049dab10c208817c05cf7/src/keyboard/textinput.js#L648) used to detect keyboard arrow presses for ipad. I think it is not needed anymore, but i was not able to test, so left it as is for now.

**Comment by InspiredGuy** [comment]
Might be also related to https://github.com/ajaxorg/ace/issues/4304 - could you please confirm if this PR fixes this issue as well?

**Review by andrewnester — APPROVED** [review]

**Comment by nightwing** [comment]
This pr does not solve that issue, which is about mouse selection handles.

**Pull request merged**